### PR TITLE
{Cognitive Services} Add private endpoint support for Cognitive Services.

### PR DESCRIPTION
**Pull request by yangyuan**
**Description**  
Add private endpoint support for Cognitive Services.

**Testing Guide**  

**History Notes**  
{Cognitive Services} Add private endpoint support for Cognitive Services.---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [x] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [x] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

### CODE REVIEW

**Comment by MyronFanQiu** [comment]
@yangyuan Should the title be `[Cognitive Services]`?

**Comment by yonzhan** [comment]
Cognitive Services

**Comment by yangyuan** [comment]
@MyronFanQiu I am not sure, because no really command change or anything. I think no need for announcing in history.

File: src/azure-cli/azure/cli/command_modules/cognitiveservices/tests/latest/test_private_endpoint.py [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+        self.kwargs.update({
+            'accountId': account['id']
+        })
+
+        self.cmd('az network vnet subnet update --name default --resource-group {rg} --vnet-name {vnetname} --disable-private-endpoint-network-policies true')
+        self.cmd('az network private-endpoint create -g {rg} -n {pename} --vnet-name {vnetname} --subnet default --private-connection-resource-id {accountId} --group-id account --connection-name {pename} -l {location}')
+
+        account = self.cmd('az cognitiveservices account show -n {sname} -g {rg}').get_output_in_json()
+        self.assertTrue(len(account['properties']['privateEndpointConnections']) > 0)
+        self.assertTrue(account['properties']['privateEndpointConnections'][0]['properties']['privateLinkServiceConnectionState']['status'], 'Approved')
```
Comment (MyronFanQiu) [inline src/azure-cli/azure/cli/command_modules/cognitiveservices/tests/latest/test_private_endpoint.py:58]: we should add test for `approve`, `reject` and `list` commands

Comment (MyronFanQiu) [reply src/azure-cli/azure/cli/command_modules/cognitiveservices/tests/latest/test_private_endpoint.py:58]: Based on our discussion with network team, we want to have a central command to manage all private endpoint resources. Which properties does CS RP need?

Comment (yangyuan) [reply src/azure-cli/azure/cli/command_modules/cognitiveservices/tests/latest/test_private_endpoint.py:58]: @MyronFanQiu contrary to what I said, network CLI actually bring extra fields to our RP and our RP cannot recognize.
We are fixing our RP and will deploy this week, after that I'll update this RP to contains `approve`, `reject` etc commands.

**Comment by yonzhan** [comment]
add to S172

**Comment by MyronFanQiu** [comment]
@yangyuan Hey. What's the status of this PR?

**Comment by MyronFanQiu** [comment]
@yangyuan Kindly ping.

**Comment by yangyuan** [comment]
Hi @MyronFanQiu, sorry for late response. I met some issue releasing RP in the past month and we just got released today.

I've updated the test with PEC approve/reject/delete/list commands. And all works .

**Comment by MyronFanQiu** [comment]
@yangyuan Hi. Thanks for contribution. Could you please resolve the conflicts?

**Comment by yangyuan** [comment]
> @yangyuan Hi. Thanks for contribution. Could you please resolve the conflicts?

Done :)

**Comment by MyronFanQiu** [comment]
@yangyuan Hi. Did you rerun the tests lively? Looks like recording file doesn't match your tests. Could you please rerun it and fix CI style as well?

**Comment by yangyuan** [comment]
@MyronFanQiu it can pass in the branch but cannot after merging to dev. looks like network team updated the command and my record was from an older version of network cli and causes the issue. I'll do a rebase and re-record.

**Comment by MyronFanQiu** [comment]
> @MyronFanQiu it can pass in the branch but cannot after merging to dev. looks like network team updated the command and my record was from an older version of network cli and causes the issue. I'll do a rebase and re-record.

Yes. We bump a api-version. Re-recording would solve this issue.

**Comment by MyronFanQiu** [comment]
Thanks for your contribution! 😄

**Comment by yangyuan** [comment]
@MyronFanQiu looks like all passing, one shows "in process", but it looks passed in DevOps

**Pull request merged**